src/types/runtime.go

```diff
@@ -19,19 +19,31 @@ type ZarfCommonOptions struct {
-	PackagePath  string            `json:"packagePath" jsonschema:"description=Location where a Zarf package to deploy can be found"`
-	Components   string            `json:"components" jsonschema:"description=Comma separated list of optional components to deploy"`
-	SGetKeyPath  string            `json:"sGetKeyPath" jsonschema:"description=Location where the public key component of a cosign key-pair can be found"`
-	SetVariables map[string]string `json:"setVariables" jsonschema:"description=Key-Value map of variable names and their corresponding values that will be used to template manifests and files in the Zarf package"`
+	Shasum        string            `json:"shasum" jsonschema:"description=The SHA256 checksum of the package to deploy"`
+	PackagePath   string            `json:"packagePath" jsonschema:"description=Location where a Zarf package to deploy can be found"`
+	Components    string            `json:"components" jsonschema:"description=Comma separated list of optional components to deploy"`
+	SGetKeyPath   string            `json:"sGetKeyPath" jsonschema:"description=Location where the public key component of a cosign key-pair can be found"`
+	SetVariables  map[string]string `json:"setVariables" jsonschema:"description=Key-Value map of variable names and their corresponding values that will be used to template manifests and files in the Zarf package"`
+	PublicKeyPath string            `json:"publicKeyPath" jsonschema:"description=Location where the public key component of a cosign key-pair can be found"`
 }
 
 // ZarfPublishOptions tracks the user-defined preferences during a package publish.
 type ZarfPublishOptions struct {
-	Reference   registry.Reference `jsonschema:"description=Remote registry reference"`
-	CopyOptions oras.CopyOptions   `jsonschema:"description=Options for the copy operation"`
-	PackOptions oras.PackOptions   `jsonschema:"description=Options for the pack operation"`
-	PackagePath string             `json:"packagePath" jsonschema:"description=Location where a Zarf package to publish can be found"`
+	Reference          registry.Reference `jsonschema:"description=Remote registry reference"`
+	CopyOptions        oras.CopyOptions   `jsonschema:"description=Options for the copy operation"`
+	PackOptions        oras.PackOptions   `jsonschema:"description=Options for the pack operation"`
+	PackagePath        string             `json:"packagePath" jsonschema:"description=Location where a Zarf package to publish can be found"`
+	SigningKeyPassword string             `json:"signingKeyPassword" jsonschema:"description=Password to the private key signature file that will be used to sign the published package"`
+	SigningKeyPath     string             `json:"signingKeyPath" jsonschema:"description=Location where the private key component of a cosign key-pair can be found"`
+}
+
+// ZarfPublishOptions tracks the user-defined preferences during a package publish.
```

Comment:
comment on exported type ZarfPullOptions should be of the form "ZarfPullOptions ..." (with optional leading article)

Reply:
Will approve anyways but be careful about these linting issues - we will need to run a PR to clean them up.